pkg/services/sqldb/provision.go

```diff
@@ -221,9 +226,8 @@ func (d *dbOnlyManager) preProvision(
 	if err != nil {
 		return nil, err
 	}
-
+	dt.ARMDeploymentName = uuid.NewV4().String()
```

Comment:
This was a a bug. arm deployment failed because it wasn't set. Lifecycle test found it.